firebase-installations/src/main/java/com/google/firebase/installations/FirebaseInstallations.java

```diff
@@ -103,4 +125,129 @@ String getApplicationId() {
   String getName() {
     return firebaseApp.getName();
   }
+
+  /**
+   * Get firebase installation id for the {@link FirebaseApp}. Either from the persisted fid entry
+   * if already exists and is not error state or create a new Fid on the FIS Servers and persist in
+   * the shared prefs.
+   *
+   * <pre>
```

Comment:
I love diagrams, but I'm not sure how much value this is bringing given it seems to be three parallel cases